include/picongpu/plugins/openPMD/openPMDWriter.hpp

```diff
@@ -1287,13 +1287,13 @@ make sure that environment variable OPENPMD_BP_BACKEND is not set to ADIOS1.
                     }
                     log<picLog::INPUT_OUTPUT>("openPMD:  (end) collect PML sizes for %1%") % name;
 
-                    fieldsGlobalSizeDims = pmacc::math::UInt64<simDim>::create(1);
-                    fieldsGlobalSizeDims[0] = globalSize;
-                    fieldsOffsetDims = pmacc::math::UInt64<simDim>::create(0);
-                    fieldsOffsetDims[0] = globalOffsetFile;
+                    recordGlobalSizeDims = pmacc::math::UInt64<simDim>::create(1);
+                    recordGlobalSizeDims[0] = globalSize;
+                    recordOffsetDims = pmacc::math::UInt64<simDim>::create(0);
+                    recordOffsetDims[0] = globalOffsetFile;
                 }
 
-                auto const componentSize = field_no_guard.productOfComponents();
+                auto const numDataPoints = localWindowSize.productOfComponents();
```

Comment:
I think this may be a bug. Shouldn't it use `recordLocalSizeDims` instead (difference in case we went into `if(!isDomainBound)` )?

Reply:
well catch! yes it is wrong

Reply:
I keep `recordLocalSizeDims` because `recordLocalSizeDims` is set to the same value. Never the less I write now inside of the `if(!isDomainBound)` -> `recordLocalSizeDims = precisionCast<uint64_t>(localWindowSize);` to be clear that both variables are equal, the only difference is the precision and that `recordLocalSizeDims` should be used to configure openPMD where variables named with `window` will be used to prepare local memory for dumping.

Reply:
True, at first review I didn't realize you were changing both variables accordingly.